engine/class_modules/priest/sc_priest.hpp

```diff
@@ -849,20 +852,29 @@ struct priest_spell_t : public priest_action_t<spell_t>
         priest().buffs.twist_of_fate->trigger();
       }
 
-      // Wrathful Faerie works for any direct attacks by anyone, bugging this for now
-      // TODO: maybe rework this to just be a buff that gives insanity every tick instead?
-      // https://github.com/SimCMinMax/WoW-BugTracker/issues/777
-      if ( priest().specialization() == PRIEST_SHADOW &&
-           ( s->result_type == result_amount_type::DMG_DIRECT || priest().bugs ) && s->result_amount > 0 )
+      if ( s->result_amount > 0 )
       {
         const priest_td_t* td = find_td( s->target );
-        if ( td && td->buffs.wrathful_faerie->check() )
+        // Wrathful Faerie works for any direct attacks by anyone, bugging this for now
+        // TODO: maybe rework this to just be a buff that gives insanity every tick instead?
+        // https://github.com/SimCMinMax/WoW-BugTracker/issues/777
+        if ( priest().specialization() == PRIEST_SHADOW &&
```

Comment:
is there a particular reason to check the spec here? I would assume that it is really the problem of the buffs you are checking whether they can be triggered in the first place, and not of the code here.

Reply:
i think it was a matter of not wanting to impact performance of disc/holy sims at the time. But not sure that really matters